### PR TITLE
Snowmobiles leave tracks, have headlights

### PR DESCRIPTION
**Pull request by ghost**
:cl:
 * tweak: Dragging bodies in snow or driving a snowmobile now leaves tracks instead of footprints. Flying over snow no longer generates any footprints
 * tweak: Made the snowmobile wrong floor sound a lot less obnoxious
 * rscadd: Snowmobiles now have headlights, you can turn them on to see a lot better in the darkness of winter

### CODE REVIEW

File: code/game/objects/structures/vehicles/snowmobile.dm [hunk 1]
```diff
@@ -1,18 +1,60 @@
 /obj/item/key/snowmobile
-	name = "\improper Snowmobile key"
-	desc = "Someone has to do it."
+	name = "\improper snowmobile key"
+	desc = "An ignition key for use with Nanotrasen's snowmobile brand. You can't quite make out the brand number, but with any luck it will go into any snowmobile."
```
Comment (Kurfursten) [inline code/game/objects/structures/vehicles/snowmobile.dm]: don't use this desc, this isn't the universal key

File: code/game/objects/structures/vehicles/snowmobile.dm [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
+			playsound(src, sound_off, 50, 1)
+
+/obj/structure/bed/chair/vehicle/snowmobile/verb/toggle_light()
+	set name = "Toggle headlights"
+	set category = "Object"
+	set src in view(1)
+
+	if(!usr.stat)
+		if(!isturf(usr.loc))
+			to_chat(usr, "You cannot turn the light on while in this [usr.loc].")//To prevent some lighting anomalities.
```
Comment (Kurfursten) [inline code/game/objects/structures/vehicles/snowmobile.dm]: while in \the

File: code/game/objects/structures/vehicles/snowmobile.dm [hunk 3]
```diff
@@ -1,18 +1,60 @@
...
+		set_light(0)
+		if(playsound && has_sound)
+			playsound(src, sound_off, 50, 1)
+
+/obj/structure/bed/chair/vehicle/snowmobile/verb/toggle_light()
+	set name = "Toggle headlights"
+	set category = "Object"
+	set src in view(1)
+
+	if(!usr.stat)
```
Comment (Kurfursten) [inline code/game/objects/structures/vehicles/snowmobile.dm]: also check restrained and probably dexterity

File: code/game/objects/structures/vehicles/snowmobile.dm [hunk 4]
```diff
@@ -24,24 +66,24 @@
...
+	if(is_type_in_list(T, approved_terrain))
 		return 1
 	else
-		var/list/dragsounds = list('sound/misc/metal_drag1.ogg','sound/misc/metal_drag2.ogg','sound/misc/metal_drag3.ogg')
-		playsound(src, pick(dragsounds), 50, 1)
-		return 5 //It's not designed to move this way!
\ No newline at end of file
+		var/list/dragsounds = list('sound/misc/metal_drag1.ogg', 'sound/misc/metal_drag2.ogg', 'sound/misc/metal_drag3.ogg')
+		playsound(src, pick(dragsounds), 10, 1) //The scratching sound is VERY loud, obnoxious and repeats often, so make sure the volume is low
+		return 7 //It's not designed to move this way!
```
Comment (Kurfursten) [inline code/game/objects/structures/vehicles/snowmobile.dm]: undocumented speed nerf

Comment (ghost) [reply code/game/objects/structures/vehicles/snowmobile.dm]: Oh shit that's the speed, let me fix that

File: code/game/objects/structures/vehicles/snowmobile.dm [hunk 5]
```diff
@@ -1,18 +1,60 @@
 /obj/item/key/snowmobile
-	name = "\improper Snowmobile key"
-	desc = "Someone has to do it."
+	name = "\improper snowmobile key"
```
Comment (jknpj) [inline code/game/objects/structures/vehicles/snowmobile.dm]: "snowmobile key" is already improper

**Comment by Kurfursten** [comment]
Speedmerged

**Pull request merged**